auto_compressor/src/lib.rs

```diff
@@ -50,3 +54,70 @@ impl FromStr for LevelInfo {
         Ok(LevelInfo(level_info))
     }
 }
+
+// PyO3 INTERFACE STARTS HERE
+#[pymodule]
+fn auto_compressor(_py: Python, m: &PyModule) -> PyResult<()> {
+    #[pyfn(m, compress_largest_rooms)]
+    fn compress_state_events_table(
+        py: Python,
+        db_url: String,
+        chunk_size: i64,
+        default_levels: String,
+        number_of_chunks: i64,
+    ) -> PyResult<()> {
+        // Stops the compressor from holding the GIL while running
+        py.allow_threads(|| {
+            compress_state_events_table_body(db_url, chunk_size, default_levels, number_of_chunks)
+        })
+    }
+
+    fn compress_state_events_table_body(
+        db_url: String,
+        chunk_size: i64,
+        default_levels: String,
+        number_of_chunks: i64,
+    ) -> PyResult<()> {
+        let _ = pyo3_log::Logger::default()
```

Comment:
Don't we only want to do this once when the module is imported?

Reply:
Yep! I moved it to run on module import (I don't think I knew that was possible when I wrote this originally!)